src/uu/mv/src/mv.rs

```diff
@@ -751,3 +845,370 @@ fn is_empty_dir(path: &Path) -> bool {
+                let file_type = dir_entry.file_type();
+                let dir_entry_path = dir_entry.into_path();
+                //comment why this is okay
+                let tmp_to = dir_entry_path.strip_prefix(&from).unwrap();
+                let dir_entry_to = to.join(tmp_to);
+                if file_type.is_dir() {
+                    // check this create all align with gnu
+                    let res = create_all(&dir_entry_to, false);
+                    if res.is_err() {
+                        error_occurred = true;
+                        show_error!("{:?}", res.unwrap_err());
+                        continue;
+                    }
+                    if let Some(vc) = vebose_context.as_ref() {
+                        vc.create_folder(&dir_entry_to)
+                    }
+                } else {
+                    let result_file_copy =
+                        copy_file(&dir_entry_path, &dir_entry_to, progress_bar, result);
+
+                    match result_file_copy {
+                        Ok(result_file_copy) => {
+                            result += result_file_copy;
+                        }
+                        Err(err) => {
+                            let err_msg = match err.kind {
+                                fs_extra::error::ErrorKind::Io(error) => {
+                                    format!("error writing {}: {}", dir_entry_to.quote(), error)
+                                }
+                                _ => {
+                                    format!("{:?}", err)
+                                }
+                            };
+                            error_occurred = true;
+                            show_error!("{}", err_msg);
+                            continue;
+                        }
+                    }
+                    if let Some(vc) = vebose_context.as_ref() {
+                        vc.print_copy_file(&dir_entry_path, &dir_entry_to, false);
+                    }
+                }
+                moved_entries.push((dir_entry_path, file_type));
+            }
+            Err(err) => {
+                let err_msg = match (err.io_error(), err.path()) {
+                    (Some(io_error), Some(path)) => {
+                        format!("cannot access {}: {io_error}", path.quote())
+                    }
+                    _ => err.to_string(),
+                };
+                show_error!("{err_msg}");
+                error_occurred = true;
+            }
+        }
+    }
+    if !error_occurred {
+        while !moved_entries.is_empty() {
+            let (moved_path, file_type) = moved_entries.pop().unwrap();
+            let res = if file_type.is_dir() {
+                remove(&moved_path)
+            } else {
+                file::remove(&moved_path)
+            };
+            if let Err(err) = res {
+                error_occurred = true;
+                show_error!("cannot remove {}: {}", moved_path.quote(), err);
```

Comment:
Please check how GNU `mv` behaves and add a test:
- Does it abort on the first `rm` error, e.g. if the file is readable but cannot be removed for some reason?
- Does it continue removing the rest?
- If a single file in a nested directory cannot get removed, does this result in multiple error messages, or just one?